### PR TITLE
【KernelGen】Add nonzero_numpy operator

### PR DESCRIPTION
**Pull request by Schopenhauer-loves-Hegel**
### PR Category
Operator

### Type of Change
New Feature

### Description
Add `nonzero_numpy` operator implementation with Triton kernel.

- Implementation mode: `manual_kernel`
- Accuracy test: 24/24 passed

### Issue
N/A

### Progress

- [ ] Change is properly reviewed (1 reviewer required, 2 recommended).
- [ ] Change is responded to an issue.
- [x] Change is fully covered by a UT.

### Performance
**torch.bfloat16**

| Shape | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|-------|-------------------:|-------------------:|--------:|
| [64, 64] | 0.0604 | 0.2958 | 0.204 |
| [256, 256] | 0.0634 | 0.3826 | 0.166 |
| [1024, 1024] | 0.0667 | 0.3940 | 0.169 |
| [4096, 4096] | 0.5237 | 0.8263 | 0.634 |
| [1024, 65536] | 1.9437 | 2.3741 | 0.819 |
| [10000, 1] | 0.0641 | 0.3143 | 0.204 |
| [10000, 256] | 0.1064 | 0.4239 | 0.251 |
| [10000, 65536] | 18.6106 | 21.4398 | 0.868 |

**torch.bool**

| Shape | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|-------|-------------------:|-------------------:|--------:|
| [64, 64] | 0.0595 | 0.2004 | 0.297 |
| [256, 256] | 0.0638 | 0.2939 | 0.217 |
| [1024, 1024] | 0.0643 | 0.2924 | 0.220 |
| [4096, 4096] | 0.3146 | 0.5887 | 0.534 |
| [1024, 65536] | 1.1322 | 1.6725 | 0.677 |
| [10000, 1] | 0.0635 | 0.2020 | 0.315 |
| [10000, 256] | 0.0772 | 0.3083 | 0.250 |
| [10000, 65536] | 10.6116 | 14.5354 | 0.730 |

**torch.float16**

| Shape | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|-------|-------------------:|-------------------:|--------:|
| [64, 64] | 0.0622 | 0.2987 | 0.208 |
| [256, 256] | 0.0631 | 0.3833 | 0.165 |
| [1024, 1024] | 0.0659 | 0.3975 | 0.166 |
| [4096, 4096] | 0.5262 | 0.8275 | 0.636 |
| [1024, 65536] | 1.9399 | 2.3785 | 0.816 |
| [10000, 1] | 0.0633 | 0.2943 | 0.215 |
| [10000, 256] | 0.1063 | 0.4253 | 0.250 |
| [10000, 65536] | 18.5689 | 21.4910 | 0.864 |

**torch.float32**

| Shape | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|-------|-------------------:|-------------------:|--------:|
| [64, 64] | 0.0608 | 0.2934 | 0.207 |
| [256, 256] | 0.0633 | 0.3826 | 0.166 |
| [1024, 1024] | 0.0750 | 0.3994 | 0.188 |
| [4096, 4096] | 0.5718 | 0.8310 | 0.688 |
| [1024, 65536] | 2.1408 | 2.4800 | 0.863 |
| [10000, 1] | 0.0637 | 0.2936 | 0.217 |
| [10000, 256] | 0.1134 | 0.4293 | 0.264 |
| [10000, 65536] | 20.6683 | 22.3866 | 0.923 |

**torch.int16**

| Shape | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|-------|-------------------:|-------------------:|--------:|
| [64, 64] | 0.0622 | 0.2935 | 0.212 |
| [256, 256] | 0.0625 | 0.3829 | 0.163 |
| [1024, 1024] | 0.0651 | 0.3921 | 0.166 |
| [4096, 4096] | 0.5244 | 0.8313 | 0.631 |
| [1024, 65536] | 1.9430 | 2.3822 | 0.816 |
| [10000, 1] | 0.0642 | 0.2934 | 0.219 |
| [10000, 256] | 0.1073 | 0.4237 | 0.253 |
| [10000, 65536] | 18.5533 | 21.4803 | 0.864 |

**torch.int32**

| Shape | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|-------|-------------------:|-------------------:|--------:|
| [64, 64] | 0.0596 | 0.2933 | 0.203 |
| [256, 256] | 0.0629 | 0.3884 | 0.162 |
| [1024, 1024] | 0.0665 | 0.3933 | 0.169 |
| [4096, 4096] | 0.5714 | 0.8205 | 0.696 |
| [1024, 65536] | 2.1585 | 2.4437 | 0.883 |
| [10000, 1] | 0.0637 | 0.2925 | 0.218 |
| [10000, 256] | 0.1135 | 0.4233 | 0.268 |
| [10000, 65536] | 20.5569 | 22.0224 | 0.933 |

**Overall: median speedup = 0.251x, mean speedup = 0.422x** (48 data points)

---
_Generated by auto_gen tool with Claude Code_

### CODE REVIEW

**Review by tengqm — DISMISSED** [review]

File: benchmark/test_nonzero_numpy_perf.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import pytest
```
Comment (tengqm) [inline benchmark/test_nonzero_numpy_perf.py]: This file is under the `benchmark` directory, so the `_perf` word in the file name is purely redundant and meaningless ... please remove.

File: benchmark/test_nonzero_numpy_perf.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import torch
+
+from benchmark.attri_util import BOOL_DTYPES, FLOAT_DTYPES, INT_DTYPES
+from benchmark.performance_utils import GenericBenchmark, unary_input_fn
+
+
+@pytest.mark.nonzero_numpy
+def test_perf_nonzero_numpy():
```
Comment (tengqm) [inline benchmark/test_nonzero_numpy_perf.py]: The same applies to function name ... all test cases (files/functions) in this directory is about performance benchmark. Having `_perf` in the function name is redundant and useless.

```suggestion
def test_nonzero_numpy():
```

File: tests/test_nonzero_numpy.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    INT_DTYPES,
+    NONZERO_SHAPES,
+    gems_assert_equal,
+    to_reference,
+)
+
+
+@pytest.mark.nonzero_numpy
+@pytest.mark.parametrize("shape", NONZERO_SHAPES)
+@pytest.mark.parametrize("dtype", FLOAT_DTYPES + INT_DTYPES + [torch.bool])
```
Comment (tengqm) [inline tests/test_nonzero_numpy.py]: [torch.bool] -> [BOOL_TYPE] ?

File: tests/test_nonzero_numpy.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        inp = torch.randn(shape, dtype=dtype, device=flag_gems.device)
+    ref_inp = to_reference(inp, False)
+
+    ref_out = torch.ops.aten.nonzero_numpy(ref_inp)
+    with flag_gems.use_gems():
+        res_out = torch.ops.aten.nonzero_numpy(inp)
+
+    assert len(res_out) == len(ref_out), "Number of output tensors should match"
+    for res_t, ref_t in zip(res_out, ref_out):
+        gems_assert_equal(res_t, ref_t)
```
Comment (tengqm) [inline tests/test_nonzero_numpy.py:38]: There are float numbers in the input, which cannot be asserted using `assert_equal`.

**Comment by factnn** [comment]
## UT Results

Branch rebased to upstream/master and tests passed:

```
python -m pytest tests/ -m nonzero --ref cpu -x -q
18 passed
```

**Comment by w1120029931-bit** [comment]
please resolve comments

**Comment by factnn** [comment]
Thanks for the review! All addressed:

1. Renamed `benchmark/test_nonzero_numpy_perf.py` → `benchmark/test_nonzero_numpy.py` ✅
2. Renamed `test_perf_nonzero_numpy` → `test_nonzero_numpy` ✅
3. Already using `BOOL_TYPES` (not `[torch.bool]`) ✅
4. Re: `gems_assert_equal` — `nonzero_numpy` outputs integer index tensors (dtype=int64), not float values, so `assert_equal` is correct here. The float input only affects *which* indices are returned, not the dtype of the output.

**Review by w1120029931-bit — APPROVED** [review]
LGTM
